pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroRecordExtractor.java

```diff
@@ -39,10 +42,15 @@ public void init(Set<String> fields, @Nullable RecordExtractorConfig recordExtra
 
   @Override
   public GenericRow extract(GenericRecord from, GenericRow to) {
-    for (String fieldName : _fields) {
-      Object value = from.get(fieldName);
-      Object convertedValue = AvroUtils.convert(value);
-      to.putValue(fieldName, convertedValue);
+    if (_fields == null || _fields.isEmpty()) { // extract all
```

Comment:
since we know this in the init, can we set a final boolean _extractAll? jvm might be able to optimize it.

Reply:
done